daffodil-schematron/src/test/resources/xsd/extends-1.dfdl.xsd

```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+           xmlns:sch="http://purl.oclc.org/dsdl/schematron">
+
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <xs:annotation>
+        <xs:appinfo source="http://www.ogf.org/dfdl/">
+            <dfdl:format ref="GeneralFormat"/>
+        </xs:appinfo>
+        <xs:appinfo>
+            <sch:pattern>
+                <sch:rule abstract="true" id="blank">
+                    <sch:assert test="string-length(.) > 0"><sch:name/> is blank.</sch:assert>
```

Comment:
I expected the test expression to have prefixes like fn:string-length(.)

Is the schematron expression language not XPath based? I am also surprised to see ">" as an operation.

Reply:
XPath allows the greater than '>'  symbol, but not the less than '<' symbol

Reply:
Thanks. What about the namespaces for the function names? 

The reason I focus on this is that DFDL expressions are XPath 2.0 derived, though not identically those, but DFDL expressions the function names are in a namespace, which usually requires a namespace prefix like "fn:". 

This inconsistency is manageable, but unfortunate if expressions like these that could, in principle be either DFDL expressions or schematron expressions, can't be cut/pasted between schematron rules and say, DFDL asserts, because the expression languages are that different. 

This really only matters for very simple expressions. DFDL expressions basically can't do anything that can't execute in constant time. If it involves searching, counting, or really any sort of iteration over the infoset tree, then DFDL expressions don't allow it for the most part, and Daffodil doesn't implement it. Those are the kinds of things that you need schematron for.  So DFDL expressions can already check lengths, but a query like "all FOO elements must be beneath a BAR element" or "there must be 3 or more FOO children of each QUUX element" can't be expressed.

Reply:
All the references I see say there is a full set of operators in both xpath 1 and 2.  Here is one of several without looking too hard https://www.w3schools.com/xml/xpath_operators.asp

Curious as to what I am missing in this discussion about limited xpath operators?

Reply:
XPath 1.0 seems to not require a correct namespace on function names. XPath 2.0 does. 

per: https://www.w3.org/2006/xpath-functions/

DFDL's expression language is similar to XPath 2.0. E.g., you cannot just type string-length(../some/path) in DFDL nor XPath 2.0 and expect it to work. You must bind a prefix (usually "fn" to namespace http://www.w3.org/2005/xpath-functions/ and use the prefix when calling the functions.

If the current schematron is limited to XPath 1.0, then so be it for now. But there's a chance it is using XPath 2.0 but turning on the XPath 1.0 compatibility mode. In which case, if we could *not* do that, then the expressions would be more harmonious with the DFDL expression language.

Reply:
Setting `queryBinding="xslt2"` on the root element will enable xslt+xpath 2.0 with our current version of schematron.

Reply:
I cant yet generate a failing test even after enabling xslt 2.0.

> E.g., you cannot just type string-length(../some/path) in DFDL nor XPath 2.0 and expect it to work. You must bind a prefix (usually "fn" to namespace http://www.w3.org/2005/xpath-functions/ and use the prefix when calling the functions.

In context of xpath, are you saying "must"  because there is enforcement or because its good practice?

Perhaps a counter example example is the GIF schema that uses unprefixed functions and enables xslt 2.0.

With all that said, I think the tools are all there, the included version of schematron supports either way.